testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/oidc/LightWeightAccessTokenMapperTest.java

```diff
@@ -0,0 +1,521 @@
+import static org.keycloak.protocol.oidc.mappers.AudienceProtocolMapper.INCLUDED_CLIENT_AUDIENCE;
+import static org.keycloak.protocol.oidc.mappers.HardcodedClaim.CLAIM_VALUE;
+import static org.keycloak.protocol.oidc.mappers.LightWeightAccessTokenMapper.EXCLUDE_FROM_ACCESS_TOKEN;
+import static org.keycloak.protocol.oidc.mappers.LightWeightAccessTokenMapper.LIGHT_WEIGHT_ACCESS_TOKEN_PROVIDER_ID;
+import static org.keycloak.protocol.oidc.mappers.OIDCAttributeMapperHelper.INCLUDE_IN_ACCESS_TOKEN;
+import static org.keycloak.protocol.oidc.mappers.OIDCAttributeMapperHelper.INCLUDE_IN_INTROSPECTION;
+import static org.keycloak.protocol.oidc.mappers.PairwiseSubMapperHelper.PAIRWISE_SUB_ALGORITHM_SALT;
+import static org.keycloak.protocol.oidc.mappers.RoleNameMapper.NEW_ROLE_NAME;
+import static org.keycloak.protocol.oidc.mappers.RoleNameMapper.ROLE_CONFIG;
+import static org.keycloak.testsuite.admin.AbstractAdminTest.loadJson;
+import static org.keycloak.testsuite.auth.page.AuthRealm.TEST;
+
+@EnableFeature(value = Profile.Feature.TOKEN_EXCHANGE, skipRestart = true)
+@EnableFeature(value = Profile.Feature.LIGHT_WEIGHT_ACCESS_TOKEN, skipRestart = true)
+public class LightWeightAccessTokenMapperTest extends AbstractKeycloakTest {
+
+    @Before
+    public void clientConfiguration() {
+        ClientManager.realm(adminClient.realm("test")).clientId("test-app").directAccessGrant(true).setServiceAccountsEnabled(true);
+        ClientManager.realm(adminClient.realm("test")).clientId("resource-server").directAccessGrant(true);
+    }
+
+    @Override
+    public void addTestRealms(List<RealmRepresentation> testRealms) {
+        RealmRepresentation realm = loadJson(getClass().getResourceAsStream("/testrealm.json"), RealmRepresentation.class);
+        ClientRepresentation confApp = KeycloakModelUtils.createClient(realm, "resource-server");
+        confApp.setSecret("password");
+        confApp.setServiceAccountsEnabled(Boolean.TRUE);
+        testRealms.add(realm);
+    }
+
+    @Test
+    public void accessTokenFalseIntrospectionTrueTest() throws IOException {
+        ProtocolMappersResource protocolMappers = setProtocolMappers(false, true, true);
+
+        oauth.nonce("123456");
+        oauth.scope("address");
+        oauth.clientId("test-app");
+        OAuthClient.AccessTokenResponse response = browserLogin("password", "test-user@localhost", "password");
+        String accessToken = response.getAccessToken();
+        System.out.println("accessToken:" + accessToken);
+        assertAccessToken(oauth.verifyToken(accessToken), true);
+
+        oauth.clientId("resource-server");
+        String tokenResponse = oauth.introspectAccessTokenWithClientCredential("resource-server", "password", accessToken);
+        System.out.println("tokenResponse:" + tokenResponse);
+        assertTokenIntrospectionResponse(JsonSerialization.readValue(tokenResponse, AccessToken.class), true);
+
+        deleteProtocolMappers(protocolMappers);
```

Comment:
Is it possible to move `deleteProtocolMappers` in the "finally" block? So something like:
```
ProtocolMappersResource protocolMappers = setProtocolMappers(false, true, true);
try {
  ....
} finally {
    deleteProtocolMappers(protocolMappers);
}
```
This is good practice, which helps, so one failing test doesn't makes the whole class to fail.

It will be good to update this way in all test methods of this class (I did not updated it in my PR, so you can better track the changes I did in my PR as this one means update of all the lines of the tests).

Reply:
Moved to the "finally" block.